analyses/new-feature-template.qmd

```diff
@@ -0,0 +1,1362 @@
+```
+
+
+
+
+### 100 Largest Initial FMV Decreases
+
+```{r}
+largest_fmv_decreases <- working_data %>%
+  arrange(diff_pred_pin_initial_fmv) %>%
+  slice(1:100)
+
+create_leaflet_map(largest_fmv_decreases, "diff_pred_pin_initial_fmv", "Largest FMV Decreases")
+```
+## Largest FMV Increase no Multicards
+
+```{r}
+largest_fmv_increases <- working_data %>%
+  group_by(meta_pin) %>%
+  filter(n() == 1) %>%
+  ungroup() %>%
+  arrange(desc(diff_pred_pin_final_fmv)) %>%
+  slice(1:100)
+
+create_leaflet_map(largest_fmv_increases, "diff_pred_pin_final_fmv", "Largest FMV Increases")
+```
+```{r}
+largest_fmv_decreases <- working_data %>%
+  group_by(meta_pin) %>%
+  filter(n() == 1) %>%
+  ungroup() %>%
+  arrange(diff_pred_pin_initial_fmv) %>%
+  slice(1:100)
+
+create_leaflet_map(largest_fmv_increases, "diff_pred_pin_final_fmv", "Largest FMV Increases")
+```
+
+
+
+:::
+
+## Neighborhoods with the Highest and Lowest SHAP Values
+
+These maps identify neighborhoods where the added variable is having the largest impact on SHAP values. By selecting neighborhoods with the highest mean(absolute value), you can take a closer look at how individual houses in these neighborhoods are affected.
+
+::: panel-tabset
+```{r processing_SHAP_values}
+working_data <- working_data %>%
+  group_by(meta_nbhd_code) %>%
+  mutate(
+    mean_value = mean(abs(!!sym(paste0(params$added_variable_shap))), na.rm = TRUE)
+  ) %>%
+  ungroup()
+
+selected_neighborhoods <- working_data %>%
+  distinct(meta_nbhd_code, .keep_all = TRUE) %>%
+  arrange(mean_value) %>%
+  slice(c(1:2, (n() - 1):n())) %>%
+  pull(meta_nbhd_code)
+
+# Filter the data to include only the selected neighborhoods
+selected_data <- working_data %>%
+  filter(meta_nbhd_code %in% selected_neighborhoods)
+
+# Separate high and low standard deviation neighborhoods
+high_mean_data <- selected_data %>%
+  filter(meta_nbhd_code %in% selected_neighborhoods[(length(selected_neighborhoods) - 1):length(selected_neighborhoods)])
+
+low_mean_data <- selected_data %>%
+  filter(meta_nbhd_code %in% selected_neighborhoods[1:2])
+```
+
+### 2 Lowest SHAP Neighborhoods
+
+```{r}
+create_leaflet_map(low_mean_data, params$added_variable_shap, "SHAP Values")
+```
+
+### 2 Highest SHAP Neighborhoods
+
+```{r}
+create_leaflet_map(high_mean_data, params$added_variable_shap, "SHAP Values")
+```
+
+:::
+
+
+## Within Neighborhood Largest Values
+
+Disparities may also be seen at the individual house level, and may be present within neighborhoods of all types, we select the top and bottom 10 SHAP values for each neighborhood. Because neighborhoods with *very* high FMVs have comparatively higher SHAP values, we also create a relative score, which can be understood as what percentage of the houses value is determined by this variable. This allows a comparison of within-neighborhood impact, while accounting for intra-neighborhood differences.
```

Comment:
```suggestion
These maps display the top and bottom 10 SHAP values by parcel for each neighborhood. Because neighborhoods with *very* high FMVs have comparatively higher SHAP values, we also use a relative score, the percentage of a parcel's value determined by the new feature. This allows a comparison of within-neighborhood impact, while accounting for intra-neighborhood differences.
```